File: AlertDialog/MainActivity.java
package com.example.alertdialog;

import android.content.DialogInterface;
import android.content.DialogInterface.OnClickListener;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toast.makeText(this,"Toast Message",Toast.LENGTH_LONG).show();
    }
    public void save (View view){

        AlertDialog.Builder alert = new AlertDialog.Builder(this);
        alert.setTitle("Save Changes?");
        alert.setMessage("Do you want to save the changes made?");

        alert.setPositiveButton("Yes" , new DialogInterface.OnClickListener(){
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Toast.makeText(getApplicationContext(), "Saved", Toast.LENGTH_SHORT).show();

            }
        });

        alert.setNegativeButton("No", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Toast.makeText(MainActivity.this, "Not Saved", Toast.LENGTH_SHORT).show();

            }
        });

        alert.setNeutralButton("Cancel", new OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Toast.makeText(MainActivity.this,"Cancelled" ,Toast.LENGTH_SHORT);
            }
        });

        alert.show();
    }
}

File: AlertView_design/alertview/MainActivity.kt
package com.gulcihan.alertview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.gulcihan.alertview.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.button1.setOnClickListener {
            val dialog = AlertDialog.Builder(this@MainActivity)

            dialog.setMessage("Message")
            dialog.setTitle("Alert")
            dialog.setIcon(R.drawable.image)

            dialog.setPositiveButton("Ok"){ _, _ ->
                Toast.makeText(applicationContext , "Okay",Toast.LENGTH_SHORT).show()
            }

            dialog.setNegativeButton("Close") { _, _ ->
                Toast.makeText(applicationContext , "Closed",Toast.LENGTH_SHORT).show()

            }
            dialog.create().show() //show AlertDialog
        }

        binding.button2.setOnClickListener {
            val design = layoutInflater.inflate(R.layout.activity_alert_view_design,null)
            val editTextAlert = design.findViewById(R.id.editTextAlert) as EditText

            val dialog = AlertDialog.Builder(this@MainActivity)

            dialog.setMessage("Message")
            dialog.setTitle("Special button")
            dialog.setIcon(R.drawable.image)

            dialog.setView(design)

            dialog.setPositiveButton("Save"){ _,_ ->
                val incomingData = editTextAlert.text.toString()

                Toast.makeText(applicationContext , "Saved : $incomingData" , Toast.LENGTH_SHORT).show()
            }

            dialog.setNegativeButton("Cancel"){_,_ ->
                Toast.makeText(applicationContext , "Cancelled" , Toast.LENGTH_SHORT).show()
            }

            dialog.create().show()
        }


    }
}